code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -229,9 +227,24 @@
 		ejectpai()
 	QDEL_NULL(internal_radio)
 	QDEL_NULL(access_card)
-	QDEL_NULL(bot_core)
 	return ..()
 
+/mob/living/simple_animal/bot/proc/check_access(mob/living/user)
+	if(user.has_unlimited_silicon_privilege || isAdminGhostAI(user)) // Silicon and Admins always have access.
+		return TRUE
+	if(!maints_access_required) // No requirements to access it.
+		return TRUE
+
+	var/obj/item/card/id/id_card = user.get_idcard(TRUE)
+	if(!id_card || !id_card.access)
+		return FALSE
+	id_card = id_card.GetID()
+
+	for(var/requested_access in maints_access_required)
+		if(requested_access in id_card.access)
+			return TRUE
+	return FALSE
```

Comment:
feels a bit similar to object's check_access_list, but I suppose it is distinct enough